### ISSUE TITLE
llava_trainer.py: Type and Attribute Error

### ISSUE THREAD

**Issue by asad14053** [VectorSpaceLab/Video-XL#8]
1) 
/Video-XL/videoxl/videoxl/train/llava_trainer.py", line 252, in compute_loss
    if "retrieval_span" in inputs:
TypeError: argument of type 'NoneType' is not iterable
Traceback (most recent call last):

2)
Video-XL/videoxl/videoxl/train/llava_trainer.py", line 268, in _prepare_inputs
    inputs.pop("length", None)
AttributeError: 'NoneType' object has no attribute 'pop'
Traceback (most recent call last):

May I know what is the fix for those error?

**Comment by shuyansy** [comment]
It seems like your input_id is None. I guess there must sth wrong with your input, could you check your training data format or show me some case of  your training data?

**Comment by asad14053** [comment]
> It seems like your input_id is None. I guess there must sth wrong with your input, could you check your training data format or show me some case of your training data?

This is my input json format, I have exactly followed https://github.com/VectorSpaceLab/Video-XL/blob/main/assets/train_example.json this format.

  {
    "id": "1",
    "video": "chicken_1.avi",
    "conversations": [
      {
        "from": "human",
        "value": "<image>\nQuestion: What are the main activities that take place in the video?"
      },
      {
        "from": "gpt",
        "value": "A video featuring a surveillance camera monitoring three chicken cages...."
      }
    ]
  }
  
  Is there any other input formats?

**Comment by shuyansy** [comment]
Thanks for providing the sample. Could you check is "<image>\n" exists in {"from":"human","value":} pairs?

**Comment by asad14053** [comment]
> Thanks for providing the sample. Could you check is "\n" exists in {"from":"human","value":} pairs?

Yes, '\n' exist **only** in {"from":"human","value":} pairs for every input. Is that, what you mean?

**Comment by shuyansy** [comment]
Actually I mean the image token like this. Meanwhile, do you use all the data I provided?
![截屏2024-10-31 上午10 12 44](https://github.com/user-attachments/assets/ea7db3cd-6a9d-42c2-9d49-b6e599a74940)

**Comment by asad14053** [comment]
> Actually I mean the image token like this. Meanwhile, do you use all the data I provided? ![截屏2024-10-31 上午10 12 44](https://private-user-images.githubusercontent.com/50947689/381798954-ea7db3cd-6a9d-42c2-9d49-b6e599a74940.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzNTE1MDgsIm5iZiI6MTczMDM1MTIwOCwicGF0aCI6Ii81MDk0NzY4OS8zODE3OTg5NTQtZWE3ZGIzY2QtNmE5ZC00MmMyLTlkNDktYjZlNTk5YTc0OTQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMxVDA1MDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmNjU4NDJhMmQ1NmRlMzMwN2EwZmY1NDBjMjgyNWFlMTk0MjhjZTkyZGM3ZWRmZmVkZjg2MDI2N2NiZjg2OWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VLwJfFP9EsuNWFLbVxUTqDfq3RP2Oovg6uklFTg4uUw)

Yes, this format, I applied for all input inference in my fine-tuning.

**Comment by asad14053** [comment]
> OK, I will try finetune on other machines. And I will upload some data I used within these days. At that time, maybe you can first use my data to train. If it is ok, you can debug what is wrong with your custom data.

Thanks for your reply.
Can you explain once more the input format for video data fine tuning?

**Comment by shuyansy** [comment]
I  think you can check these part of codes(videoxl/videoxk/model/llava_arch.py). For the video input, the tensor shape of video should be (N,3,144,144); where N is the frame of videos.

def encode_multimodals(self, videos_or_images, video_idx_in_batch, split_sizes=None):
        videos_or_images_features = self.get_model().get_vision_tower()(videos_or_images)
        per_videos_or_images_features = torch.split(videos_or_images_features, split_sizes, dim=0)  # tuple, (dim_1, 576, 4096)
        all_videos_or_images_features = []

        for idx, feat in enumerate(per_videos_or_images_features):

            feat = self.get_model().mm_projector(feat)
            # Post pooling
            if idx in video_idx_in_batch:
                feat = self.get_2dPool(feat)
            all_videos_or_images_features.append(feat)
        return all_videos_or_images_features

**Comment by shuyansy** [comment]
Hi, I am not sure the issue has been settled. I have released some training data and you can use it to fine-tune the model. Thanks once again for patient waiting.
